azkaban-execserver/src/main/java/azkaban/execapp/ServerStatisticsServlet.java

```diff
@@ -67,82 +72,116 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
-          // find the total memory and value the variable.
-          if (result.contains("MemTotal") && result.split("\\s+").length > 2){
-            try {
-              totalMemory = Long.parseLong(result.split("\\s+")[1]);
-              logger.info("Total memory : " + totalMemory);
-            }catch(NumberFormatException e){
-              logger.error("yielding 0 for total memory as output is invalid -" + result);
+        long totalMemory = 0;
+        long totalFreeMemory = 0;
+        Long parsedResult = (long) 0;
+
+        // process the output from bash call.
+        // we expect the result from the bash call to be something like following -
+        // MemTotal:       65894264 kB
+        // MemFree:        57753844 kB
+        // Buffers:          305552 kB
+        // Cached:          3802432 kB
+        // SwapCached:            0 kB
+        // Note : total free memory = freeMemory + cached + buffers + swapCached
+        // TODO : think about merging the logic in systemMemoryInfo as the logic is similar
+        if (output.size() == 5) {
+          for (String result : output) {
+            // find the total memory and value the variable.
+            parsedResult = extractMemoryInfo("MemTotal", result);
+            if (null != parsedResult) {
+              totalMemory = parsedResult;
+              continue;
             }
-          }
-          // find the free memory and value the variable.
-          if (result.contains("MemFree") && result.split("\\s+").length > 2){
-            try {
-              freeMemory = Long.parseLong(result.split("\\s+")[1]);
-              logger.info("Free memory : " + freeMemory);
-            }catch(NumberFormatException e){
-              logger.error("yielding 0 for total memory as output is invalid -" + result);
+
+            // find the free memory.
+            parsedResult = extractMemoryInfo("MemFree", result);
+            if (null != parsedResult) {
+              totalFreeMemory += parsedResult;
+              continue;
+            }
+
+            // find the Buffers.
+            parsedResult = extractMemoryInfo("Buffers", result);
+            if (null != parsedResult) {
+              totalFreeMemory += parsedResult;
+              continue;
+            }
+
+            // find the Cached.
+            parsedResult = extractMemoryInfo("SwapCached", result);
+            if (null != parsedResult) {
+              totalFreeMemory += parsedResult;
+              continue;
+            }
+
+            // find the Cached.
+            parsedResult = extractMemoryInfo("Cached", result);
+            if (null != parsedResult) {
+              totalFreeMemory += parsedResult;
+              continue;
             }
           }
+        } else {
+          logger.error("failed to get total/free memory info as the bash call returned invalid result."
+              + String.format(" Expected lines of output = 5, actual = %s ", output.size()));
```

Comment:
might help to print which actual values you got? (i.e. which of the 3 you got, for instance)